### DISCUSSION TITLE
Not able to build knowledgeMW in Build core

### DISCUSSION THREAD

**Discussion by nikhilvarma940** [Sunbird-Ed/Community#486] [Installation] [answered]
Getting the below error in console output.please help me with the issue.

Started by user admin
Rebuilds build #3
Checking out git https://github.com/project-sunbird/knowledge-mw-service.git into /var/lib/jenkins/workspace/Build/Core/KnowledgeMW@script/cf2d4f90bd1971d1866eb7637deb5a4f1df2d06c248104d0c1267bfea70f431e to read Jenkinsfile
The recommended git tool is: NONE
No credentials specified
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/Build/Core/KnowledgeMW@script/cf2d4f90bd1971d1866eb7637deb5a4f1df2d06c248104d0c1267bfea70f431e/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/project-sunbird/knowledge-mw-service.git # timeout=10
Fetching upstream changes from https://github.com/project-sunbird/knowledge-mw-service.git
 > git --version # timeout=10
 > git --version # 'git version 2.34.1'
 > git fetch --tags --force --progress -- https://github.com/project-sunbird/knowledge-mw-service.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse origin/release-5.2.0_RC1^{commit} # timeout=10
 > git rev-parse release-5.2.0_RC1^{commit} # timeout=10
Checking out Revision d110f8f90da1f7ee020f0f03383306ded0dbef14 (release-5.2.0_RC1)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f d110f8f90da1f7ee020f0f03383306ded0dbef14 # timeout=10
Commit message: "Issue #KN-677 merge: Merge pull request #297 from HarishGangula/release-5.3.0"
 > git rev-list --no-walk d110f8f90da1f7ee020f0f03383306ded0dbef14 # timeout=10
[Pipeline] Start of Pipeline
[Pipeline] node
Running on Jenkins in /var/lib/jenkins/workspace/Build/Core/KnowledgeMW
[Pipeline] {
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (Checkout)
[Pipeline] echo
[1m[32mFound environment variable named hub_org with value as: nikhilvarma940[0m
[Pipeline] cleanWs
[WS-CLEANUP] Deleting project workspace...
[WS-CLEANUP] Deferred wipeout is used...
[WS-CLEANUP] done
[Pipeline] checkout
The recommended git tool is: NONE
No credentials specified
Cloning the remote Git repository
Cloning repository https://github.com/project-sunbird/knowledge-mw-service.git
 > git init /var/lib/jenkins/workspace/Build/Core/KnowledgeMW # timeout=10
Fetching upstream changes from https://github.com/project-sunbird/knowledge-mw-service.git
 > git --version # timeout=10
 > git --version # 'git version 2.34.1'
 > git fetch --tags --force --progress -- https://github.com/project-sunbird/knowledge-mw-service.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/project-sunbird/knowledge-mw-service.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
 > git rev-parse origin/release-5.2.0_RC1^{commit} # timeout=10
 > git rev-parse release-5.2.0_RC1^{commit} # timeout=10
Checking out Revision d110f8f90da1f7ee020f0f03383306ded0dbef14 (release-5.2.0_RC1)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f d110f8f90da1f7ee020f0f03383306ded0dbef14 # timeout=10
Commit message: "Issue #KN-677 merge: Merge pull request #297 from HarishGangula/release-5.3.0"
[Pipeline] sh
+ git rev-parse --short HEAD
[Pipeline] sh
+ echo release-5.2.0_RC1_d110f8f_4
[Pipeline] echo
build_tag: release-5.2.0_RC1_d110f8f_4
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Build)
[Pipeline] echo
Environment will be : build
[Pipeline] sh
+ git submodule update --init
Submodule 'src/libs' (https://github.com/project-sunbird/sunbird-js-utils) registered for path 'src/libs'
Cloning into '/var/lib/jenkins/workspace/Build/Core/KnowledgeMW/src/libs'...
Submodule path 'src/libs': checked out '74075c8d3354ad4b6b26f8449ea309355630582c'
[Pipeline] sh
+ chmod 777 build.sh
[Pipeline] sh
+ ./build.sh release-5.2.0_RC1_d110f8f_4 built-in nikhilvarma940
#1 [internal] load .dockerignore
#1 transferring context: 2B done
#1 DONE 0.0s

#2 [internal] load build definition from Dockerfile
#2 transferring dockerfile: 524B done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/circleci/node:8.11.2-stretch
#3 DONE 1.8s

#4 [internal] load metadata for docker.io/library/node:8.11-slim
#4 DONE 2.6s

#5 [stage-0 1/4] FROM docker.io/circleci/node:8.11.2-stretch@sha256:147dd7f8267b50bb827a9682bcd774c0923e03c82c2a8bbfa303bd36b7304c74
#5 DONE 0.0s

#6 [stage-1 1/4] FROM docker.io/library/node:8.11-slim@sha256:682383b9e173828b786e3d3513739e9280492d3ea249655b03753dfc3bd0111d
#6 CACHED

#7 [internal] load build context
#7 transferring context: 1.67MB 0.0s done
#7 DONE 0.1s

#8 [stage-1 2/4] RUN sed -i '/jessie-updates/d' /etc/apt/sources.list     && apt update     && apt-get clean     && useradd -m sunbird
#8 ...

#9 [stage-0 2/4] COPY src /opt/content/
#9 CACHED

#10 [stage-0 3/4] WORKDIR /opt/content/
#10 CACHED

#11 [stage-0 4/4] RUN npm install --unsafe-perm
#11 1.308 
#11 1.308 > content-service@1.14.0 preinstall /opt/content
#11 1.308 > npm install ./libs/sb-config-util && npm install ./libs/sb-http-util && npm install ./libs/sb_content_provider_util && npm install ./libs/sb_logger_util && npm install ./libs/sb_logger_util_v2 && npm install ./libs/sb_req_validator_util && npm install ./libs/response_util && npm install ./libs/sb_api_interceptor && npm install ./libs/sb_cache_manager && npm install ./libs/sb_telemetry_util
#11 1.308 
#11 3.326 npm WARN notice [SECURITY] lodash has the following vulnerability: 1 high. Go here for more details: https://github.com/advisories?query=lodash - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#11 3.344 npm WARN notice [SECURITY] lodash has the following vulnerabilities: 1 critical, 3 high, 2 moderate, 1 low. Go here for more details: https://github.com/advisories?query=lodash - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#11 3.362 npm WARN notice [SECURITY] ansi-regex has the following vulnerability: 1 moderate. Go here for more details: https://github.com/advisories?query=ansi-regex - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#11 3.615 npm WARN notice [SECURITY] lodash has the following vulnerabilities: 1 critical, 3 high, 1 moderate, 1 low. Go here for more details: https://github.com/advisories?query=lodash - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#11 3.615 npm WARN notice [SECURITY] jsonwebtoken has the following vulnerabilities: 3 moderate. Go here for more details: https://github.com/advisories?query=jsonwebtoken - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#11 4.200 npm WARN notice [SECURITY] karma has the following vulnerabilities: 2 moderate. Go here for more details: https://github.com/advisories?query=karma - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#11 4.219 npm WARN notice [SECURITY] socket.io has the following vulnerability: 1 moderate. Go here for more details: https://github.com/advisories?query=socket.io - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#11 4.717 npm WARN notice [SECURITY] xmlhttprequest-ssl has the following vulnerabilities: 1 critical, 1 high. Go here for more details: https://github.com/advisories?query=xmlhttprequest-ssl - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#11 4.776 npm WARN notice [SECURITY] engine.io has the following vulnerability: 1 high. Go here for more details: https://github.com/advisories?query=engine.io - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#11 4.795 npm WARN notice [SECURITY] xml2js has the following vulnerability: 1 moderate. Go here for more details: https://github.com/advisories?query=xml2js - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#11 6.180 npm WARN notice [SECURITY] log4js has the following vulnerability: 1 moderate. Go here for more details: https://github.com/advisories?query=log4js - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
#11 ...

#8 [stage-1 2/4] RUN sed -i '/jessie-updates/d' /etc/apt/sources.list     && apt update     && apt-get clean     && useradd -m sunbird
#8 0.562 
#8 0.562 WARNING: apt does not have a stable CLI interface yet. Use with caution in scripts.
#8 0.562 
#8 1.950 Ign http://security.debian.org jessie/updates InRelease
#8 2.095 Ign http://security.debian.org jessie/updates Release.gpg
#8 2.099 Ign http://deb.debian.org jessie InRelease
#8 2.290 Ign http://deb.debian.org jessie Release.gpg
#8 2.365 Ign http://security.debian.org jessie/updates Release
#8 2.483 Ign http://deb.debian.org jessie Release
#8 2.517 Err http://security.debian.org jessie/updates/main amd64 Packages
#8 2.517   
#8 2.661 Err http://security.debian.org jessie/updates/main amd64 Packages
#8 2.661   
#8 2.679 Err http://deb.debian.org jessie/main amd64 Packages
#8 2.679   
#8 2.872 Err http://deb.debian.org jessie/main amd64 Packages
#8 2.872   
#8 2.936 Err http://security.debian.org jessie/updates/main amd64 Packages
#8 2.936   
#8 3.065 Err http://deb.debian.org jessie/main amd64 Packages
#8 3.065   
#8 3.080 Err http://security.debian.org jessie/updates/main amd64 Packages
#8 3.080   
#8 3.221 Err http://security.debian.org jessie/updates/main amd64 Packages
#8 3.221   404  Not Found [IP: 151.101.2.132 80]
#8 3.262 Err http://deb.debian.org jessie/main amd64 Packages
#8 3.262   
#8 3.469 Err http://deb.debian.org jessie/main amd64 Packages
#8 3.469   404  Not Found
#8 3.470 W: Failed to fetch http://security.debian.org/debian-security/dists/jessie/updates/main/binary-amd64/Packages  404  Not Found [IP: 151.101.2.132 80]
#8 3.470 
#8 3.470 W: Failed to fetch http://deb.debian.org/debian/dists/jessie/main/binary-amd64/Packages  404  Not Found
#8 3.470 
#8 3.470 E: Some index files failed to download. They have been ignored, or old ones used instead.
#8 ERROR: process "/bin/sh -c sed -i '/jessie-updates/d' /etc/apt/sources.list     && apt update     && apt-get clean     && useradd -m sunbird" did not complete successfully: exit code: 100

#11 [stage-0 4/4] RUN npm install --unsafe-perm
#11 CANCELED
------
 > [stage-1 2/4] RUN sed -i '/jessie-updates/d' /etc/apt/sources.list     && apt update     && apt-get clean     && useradd -m sunbird:
#8 3.221   404  Not Found [IP: 151.101.2.132 80]
#8 3.262 Err http://deb.debian.org jessie/main amd64 Packages
#8 3.262   
#8 3.469 Err http://deb.debian.org jessie/main amd64 Packages
#8 3.469   404  Not Found
#8 3.470 W: Failed to fetch http://security.debian.org/debian-security/dists/jessie/updates/main/binary-amd64/Packages  404  Not Found [IP: 151.101.2.132 80]
#8 3.470 
#8 3.470 W: Failed to fetch http://deb.debian.org/debian/dists/jessie/main/binary-amd64/Packages  404  Not Found
#8 3.470 
#8 3.470 E: Some index files failed to download. They have been ignored, or old ones used instead.
------
Dockerfile:10
--------------------
   9 |     MAINTAINER "Manojvv" "manojv@ilimi.in"
  10 | >>> RUN sed -i '/jessie-updates/d' /etc/apt/sources.list \
  11 | >>>     && apt update \
  12 | >>>     && apt-get clean \
  13 | >>>     && useradd -m sunbird
  14 |     USER sunbird
--------------------
ERROR: failed to solve: process "/bin/sh -c sed -i '/jessie-updates/d' /etc/apt/sources.list     && apt update     && apt-get clean     && useradd -m sunbird" did not complete successfully: exit code: 100
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: script returned exit code 1
Finished: FAILURE

**Comment by gandham-santhosh** [comment]
@pallakartheekreddy can you please check this? Looks like the base nodejs docker image has an issue where the debian packages are not available

**Reply by gandham-santhosh** [reply]
@pallakartheekreddy we made changes to this https://github.com/Sunbird-Knowlg/knowledge-mw-service/pull/301 
Can you please check and merge this?

**Reply by Krishnaj20** [reply]
@pallakartheekreddy @gandham-santhosh - Any update on this?

**Reply by gandham-santhosh** [reply, marked as answer]
@Krishnaj20  I see PR is merged
@nikhilvarma940  can you please rebuild from the 5.2.0 branch and check? Please mark this thread as answered if the issue is resolved

**Reply by nikhilvarma940** [reply]
@gandham-santhosh The build is successful. Thank you
Marking this as answered and closing the discussion.

**Reply by ddevadat** [reply]
@santhosh-tg i know this question is answered..but there is no branch name release-5.2.0 in https://github.com/Sunbird-Knowlg/knowledge-mw-service

**Reply by santhosh-tg** [reply]
@ddevadat the branch should be available now https://github.com/Sunbird-Knowlg/knowledge-mw-service/tree/release-5.2.0
